docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Testing Tricky Edge Cases in a TUF Client"
+author: Ivana Atanasova
+---
+
+Usually the TUF Specification creates an impression of simple and straightforward approach to address software update systems security gaps. In the next few paragraphs we'll try to convince you that the devil is in the details.
+
+With the [v1.0.0 release](https://blogs.vmware.com/opensource/2022/02/22/python-tuf-reaches-version-1-0-0/) we can say that the current reference implementation is finally in a good place, although it wouldn’t be so trustworthy without all the awesome test functionality it provides. Therein lies some interesting surprises, for the test coverage reflects use cases that wouldn’t respectively come to mind. TUF, in fact, is capable of managing some tricky business! And that’s why it provides more than just test coverage - it is a way to analyze and validate that the implementation strictly follows the TUF spec with all its corner cases and tricky details that are far from obvious.
+Before looking into them, we need to introduce the test functionality itself.
+
+
+## Some Repository Simulator Magic
+
+The great thing about [RepositorySimulator](https://github.com/theupdateframework/python-tuf/blob/develop/tests/repository_simulator.py), is it allows you to play with repository metadata by modifying it, signing and storing new roles versions, while serving older ones in the client test code. You can also simulate downloading new metadata from a remote without the need of file access or network connections, and modify expiry dates and time.
+
+When you execute a test in python-tuf, it creates a local temporary directory serving for cache, with "/metadata/..." and "/targets/..." URL paths that host metadata and targets respectively.
+
+The test suite provides you with the ability to see the "live" test repository state for debugging purposes. You just need to use the `--dump` option, so that you can see the current tests corresponding path.
+
+Let’s cite a specific example with testing expired metadata to demonstrate two of the cool things the RepositorySimulator provides:
+* Upgrade [targets](https://theupdateframework.github.io/specification/latest/#targets) to v2
+* Change [timestamp](https://theupdateframework.github.io/specification/latest/#timestamp) v2 expiry date
```

Comment:
I think the cool thing here is the bigger picture, i.e. the whole repository update story `test_expired_metadata` tells. 

In other words, I would try to emphasize on RepositorySimulator's capability to simulate real repository chains of updates as suggested by the spec, and not just modify individual metadata.